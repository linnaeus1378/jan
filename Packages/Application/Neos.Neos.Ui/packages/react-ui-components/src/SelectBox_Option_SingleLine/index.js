/* eslint-disable camelcase, react/jsx-pascal-case */
import SelectBox_Option_SingleLine from './selectBox_Option_SingleLine';

export default SelectBox_Option_SingleLine;
