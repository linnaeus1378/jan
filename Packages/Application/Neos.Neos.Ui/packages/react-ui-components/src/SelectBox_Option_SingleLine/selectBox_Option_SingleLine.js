/* eslint-disable camelcase, react/jsx-pascal-case */
import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import ListPreviewElement from '../ListPreviewElement';
import mergeClassNames from 'classnames';

class SelectBox_Option_SingleLine extends PureComponent {
    static propTypes = {
        option: PropTypes.shape({
            label: PropTypes.string.isRequired,
            icon: PropTypes.string,
            disabled: PropTypes.bool
        }).isRequired,

        disabled: PropTypes.bool,

        className: PropTypes.string
    }

    render() {
        const {option, className, disabled} = this.props;

        const isDisabled = disabled || option.disabled;

        const finalClassNames = mergeClassNames({
            [className]: className
        });

        return (
            <ListPreviewElement {...this.props} icon={option.icon} disabled={isDisabled} className={finalClassNames}>
                <span title={option.label}>{option.label}</span>
            </ListPreviewElement>
        );
    }
}

export default SelectBox_Option_SingleLine;
