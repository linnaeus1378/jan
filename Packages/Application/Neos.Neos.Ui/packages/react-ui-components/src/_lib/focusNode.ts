/**
 * Creates a curried function which can be used as a `ref` in React Components
 * to focus the rendered node.
 *
 * @param {Boolean} Indicates wether the node should be focused or not,
 */
export const makeFocusNode = (isFocused: boolean) => (node: any) => {
    if (node) {
        const method = isFocused ? 'focus' : 'blur';

        node[method]();
    }
};
