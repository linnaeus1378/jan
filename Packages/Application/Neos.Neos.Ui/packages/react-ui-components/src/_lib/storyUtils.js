import React from 'react';
import PropTypes from 'prop-types';

/**
 * The StoryWrapper is a simple component which can be used to wrap stories
 * of the react-storybook lib.
 */
export const StoryWrapper = ({children, ...rest}) => (
    <div
        style={{background: '#000', padding: '1rem', position: 'relative'}}
        {...rest}
        >
        {children}
    </div>
);
StoryWrapper.propTypes = {
    children: PropTypes.node.isRequired
};
