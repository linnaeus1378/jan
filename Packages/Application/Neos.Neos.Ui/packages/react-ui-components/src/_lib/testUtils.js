import React from 'react';

/**
 * A function which returns a simple React element which can be
 * used as a test stub for cross component dependencies.
 */
export const createStubComponent = componentName => props => <div data-component-name={componentName} {...props} style={{}}/>;
