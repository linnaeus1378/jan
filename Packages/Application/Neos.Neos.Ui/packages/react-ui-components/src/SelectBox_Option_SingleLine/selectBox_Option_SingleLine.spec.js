/* eslint-disable camelcase, react/jsx-pascal-case */
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {createStubComponent} from './../_lib/testUtils';
import SelectBox_Option_SingleLine from './selectBox_Option_SingleLine';

describe('<SelectBox_Option_SingleLine/>', () => {
    let props;

    beforeEach(() => {
        props = {
            label: 'Foo label',
            option: {label: 'Bar label'},
            onClick: jest.fn(),
            theme: {},
            ListPreviewElement: createStubComponent()
        };
    });

    it('should render correctly.', () => {
        const wrapper = shallow(<SelectBox_Option_SingleLine {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
