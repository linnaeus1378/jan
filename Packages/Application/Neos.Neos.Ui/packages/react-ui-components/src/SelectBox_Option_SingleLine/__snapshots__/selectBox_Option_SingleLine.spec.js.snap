// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectBox_Option_SingleLine/> should render correctly. 1`] = `
<ThemedListPreviewElement
  ListPreviewElement={[Function]}
  className=""
  label="Foo label"
  onClick={[MockFunction]}
  option={
    Object {
      "label": "Bar label",
    }
  }
  theme={Object {}}
>
  <span
    title="Bar label"
  >
    Bar label
  </span>
</ThemedListPreviewElement>
`;
